WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderPostDetailViewModel.kt

```diff
@@ -133,6 +159,52 @@ class ReaderPostDetailViewModel @Inject constructor(
         }
     }
 
+    fun onShowPost(blogId: Long, postId: Long) {
+        launch { getOrFetchReaderPost(blogId = blogId, postId = postId) }
+    }
+
+    private suspend fun getOrFetchReaderPost(blogId: Long, postId: Long) {
+        _uiState.value = LoadingUiState
+
+        getReaderPostFromDb(blogId = blogId, postId = postId)
+        if (post == null) {
+            when (readerFetchPostUseCase.fetchPost(blogId = blogId, postId = postId, isFeed = isFeed)) {
+                FetchReaderPostState.Success -> {
+                    getReaderPostFromDb(blogId, postId)
+                    updatePostDetailsUi()
+                }
+
+                FetchReaderPostState.AlreadyRunning -> {
+                    AppLog.i(T.READER, "reader post detail > fetch post already running")
+                    _uiState.value = ErrorUiState(null)
+                }
+
+                FetchReaderPostState.Failed.NoNetwork ->
+                    _uiState.value = ErrorUiState(UiStringRes(R.string.no_network_message))
+
+                FetchReaderPostState.Failed.RequestFailed ->
+                    _uiState.value = ErrorUiState(UiStringRes(R.string.reader_err_get_post_generic))
+
+                FetchReaderPostState.Failed.NotAuthorised -> updateAndTrackNotAuthorisedErrorState()
```

Comment:
1. Suggestion (💡): When an `and` is involved within a function name, I usually make sure to follow the order of the naming with the order of the called logic. For example, I would have named that as `trackAndUpdateNotAuthorisedErrorState()` instead.
2. Suggestion (💡): When an `and` is involved within a function name, I usually split the function into two inner functions or lines, to reduce the functions complexity and adhere to what the name function suggests, like:
```
    private fun trackAndUpdateNotAuthorisedErrorState() {
        trackNotAuthorisedState()
        _uiState.value = ErrorUiState(
                message = UiStringRes(getNotAuthorisedErrorMessageRes()),
                signInButtonVisibility = shouldOfferSignIn
        )
    }

    private fun getNotAuthorisedErrorMessageRes() = if (!shouldOfferSignIn) {
        if (interceptedUri == null) {
            R.string.reader_err_get_post_not_authorized
        } else {
            R.string.reader_err_get_post_not_authorized_fallback
        }
    } else {
        if (interceptedUri == null) {
            R.string.reader_err_get_post_not_authorized_signin
        } else {
            R.string.reader_err_get_post_not_authorized_signin_fallback
        }
    }
```

Reply:
Great suggestion, included in db48550.

Reply:
Nope, YOU are great! 🚀